app/controllers/account_controller.rb

```diff
@@ -300,20 +300,15 @@ def registration_through_invitation!
     session[:auth_source_registration] = nil
 
     if @user.nil?
-      @user = User.new(language: Setting.default_language)
-    elsif user_with_placeholder_name?(@user)
```

Comment:
Without knowing what that actually causes, it seems to me that this line should not be deleted since after the change, two different cases are merged into one.